src/t_stream.c

```diff
@@ -1433,17 +1448,47 @@ void xaddCommand(client *c) {
     signalKeyAsReady(c->db, c->argv[1], OBJ_STREAM);
 }
 
-/* XRANGE/XREVRANGE actual implementation. */
+/* XRANGE/XREVRANGE actual implementation.
+ * The 'start' and 'end' IDs are parsed as follows:
+ *   Incomplete 'start' has its sequence set to 0, and 'end' to UINT64_MAX.
+ *   "-" and "+"" mean the minimal and maximal ID values, respectively.
+ *   The "(" prefix means an open (exclusive) range, so XRANGE stream (1-0 (2-0
+ *   will match anything from 1-1 and 1-UINT64_MAX.
+ */
 void xrangeGenericCommand(client *c, int rev) {
     robj *o;
     stream *s;
     streamID startid, endid;
     long long count = -1;
     robj *startarg = rev ? c->argv[3] : c->argv[2];
     robj *endarg = rev ? c->argv[2] : c->argv[3];
-
-    if (streamParseIDOrReply(c,startarg,&startid,0) == C_ERR) return;
-    if (streamParseIDOrReply(c,endarg,&endid,UINT64_MAX) == C_ERR) return;
+    int startex = 0, endex = 0, invalid_id = 0;
+    char *startp = startarg->ptr, *endp = endarg->ptr;
+    size_t startlen = sdslen(startarg->ptr), endlen = sdslen(endarg->ptr);
+
+    /* Check for open/close (exclusive/inclusive) range prefixes. */
+    if (startlen > 1 && startp[0] == '(') {
+        startex = 1;
+    }
+    if (endlen > 1 && endp[0] == '(') {
+        endex = 1;
+    }
+    
+    /* Parse start and end IDs. */
+    startarg = createStringObject(startp+startex,startlen-startex);
+    endarg = createStringObject(endp+endex,endlen-endex);
```

Comment:
it's a bit weird to uses the "boolean" startex/endex together with the numeric startlen/endlen... i think we should just use `startlen-1` and `endlen-1`

Reply:
or maybe just decide that startex and endex are booleans and then `createStringObject(startp+1,startlen-1)`

Reply:
Refactored into a saner (hopefully) helper function.